python/simulation.py

```diff
@@ -590,13 +590,25 @@ def output_component(self, c, h5file=None):
                 self.last_eps_filename = nm
             self.output_h5_hook(nm)
 
+    def get_array(self, center, size, component=mp.Ez):
+        self.dim_sizes = np.zeros(3, dtype=np.int32)
```

Comment:
I don't understand why this is a field of `self`.  Why can't each call to `get_array` allocate new arrays to store the sizes?

Also, there should be an optional parameter `arr=None` so that the user can pass in a pre-allocated array (the code should check that it is the correct size and type … in most cases it would be obtained from a previous call to `get_array`).